### DISCUSSION TITLE
Can't push from Terminal: fatal: Authentication failed for <repository-url> <title>

### DISCUSSION THREAD

**Discussion by GarrettS** [community/community#126902] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Here's a summary of the troubleshooting steps I've attempted to address the authentication issue with Git, along with their outcomes:

1. Using macOS Keychain (osxkeychain) for Credential Storage:

Configured Git to use the macOS keychain (osxkeychain) as the credential helper.
Outcome: Git continued to prompt for authentication but failed to authenticate with stored credentials.

2. Generating and Using Personal Access Token (PAT):
Couldn't figure out how to do this.

3. Removed the local override for the credential.helper setting in the .gitconfig file.
Ensured Git used the global osxkeychain setting for credential storage.
Outcome: Git continued to prompt for authentication but failed to authenticate with stored credentials.

**Comment by YuheshPandian** [comment]
PAT is the most likely one to work in this situation.

Let me help you with PAT:

- Create a GitHub account by going to [github.com](https://github.com/) and signing up if you don't already have one.
- Open GitHub and sign in.
- In the top right corner of the page, click on your profile picture.
- Select **Settings* from the dropdown menu.
- In the left-hand menu, scroll down and click on **Developer settings**.
- Click on **Personal access tokens**.
- Click on **Generate new token**.
- Give your token a name.
- Choose the scopes or permissions you want your token to have, such as **repo** to access repositories.
- Click on **Generate token** at the bottom of the page.
- Copy the token immediately because you won't be able to see it again.
- Paste the token in a safe place, such as a password manager, to store it securely.
- Open your terminal or cmd.
- when pushing code into a repository, use the command:
    ```sh
    git push origin main 
    ```
- When asked for your password, paste the PAT.


# Alternative method

if you use Visual studio code try the UI instead of Terminal I face less issues when facing authentication issues in the UI Way. To do so click on the fourth icon over here on you VS Code.
![image](https://github.com/community/community/assets/168426680/e8aeedbc-0abc-49d5-922e-a1de996d6be8)


# Tutorials:
## Youtube - [Tutorial by ProgrammingKnowledge](https://youtu.be/ePCBuIQJAUc?feature=shared)
## Documentation - [Graphite.dev](https://graphite.dev/guides/how-to-fix-git-authentication-failed-error)
## Github Community Discussion - [29193](https://github.com/orgs/community/discussions/29193)



Thank you,
Yuhesh Pandian

**Reply by GarrettS** [reply]
Thank you

**Reply by YuheshPandian** [reply]
Your Welcome @GarrettS, hope i solved your problem.

**Reply by ebndev** [reply]
Hi @YuheshPandian, thanks for taking the time to answer!

@GarrettS if the response above helped, please feel encouraged to mark it as the answer. If not, let us know if you have further questions or post the solution if you've resolved it on your own. It would be helpful for other users who might experience this in the future 😄
